### ISSUE TITLE
Replace LineVisual by SegmentCollection

### ISSUE THREAD

**Issue by rossant** [vispy/vispy#831]
It seems like [agg support of LineVisual is very minimal currently](https://github.com/vispy/vispy/blob/master/vispy/visuals/line/line.py#L39). Maybe we could replace it by the new [SegmentCollection](https://github.com/vispy/vispy/blob/master/vispy/visuals/collections/segment_collection.py) that was ported from Glumpy?

**Comment by Eric89GXL** [comment]
Sounds pretty reasonable. Do you have time to work on it? If not, I could give it a shot at some point. @campagnola I assume #800 should be completed first?

**Comment by rossant** [comment]
No I won't have time to work on this for now...

**Comment by campagnola** [comment]
Yes, #800 should go through first.  There will be some design decisions to make around how best to adapt Collections to Visuals, but I don't think it will be terribly difficult.
